### PR TITLE
update travis build; add appveyor

### PR DESCRIPTION
**Pull request by ashnazg**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,6 @@
 language: php
-
-php:
-  - 7.1
-  - 7.2
+php: [7.1, 7.2, nightly ]
+sudo: false
 
 matrix:
```
Comment (ashnazg) [inline .travis.yml]: Unsure if leaving this `matrix` section in here actually works with the newer file format.  Build output doesn't appear to do the separate ECS and PHPSTAN builds that were done via the old format.

File: .travis.yml [hunk 2]
```diff
@@ -11,26 +9,43 @@ matrix:
...
-  - if [[ $ECS != "" ]]; then composer create-project symplify/easy-coding-standard temp/ecs; fi
+- composer install --no-interaction --prefer-dist -o
+- if [[ $ECS != "" ]]; then composer create-project symplify/easy-coding-standard temp/ecs; fi
+
+jobs:
+  include:
+    - stage: Test
+      script:
+        - vendor/bin/phpunit
+        - if [[ $ECS != "" ]]; then temp/ecs/bin/ecs check src tests; fi
```
Comment (jaapio) [inline .travis.yml]: Think you have to add a new include for these subtasks like

```
- stage: lint
   script: vendor/bin/phpstan analyse src --level max --configuration phpstan.neon;

- stage lint
  script: temp/ecs/bin/ecs check src tests;
  before_script: composer create-project symplify/easy-coding-standard temp/ecs;
```
The if statemens are just a hack to simulate stages. which are now available in travis

**Comment by ashnazg** [comment]
I think I have this one ready to review now.

File: README.md [hunk 3]
```diff
@@ -1,4 +1,13 @@
...
+[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
+[![Travis Status](https://travis-ci.org/phpDocumentor/ReflectionDocBlock.svg?branch=master)](https://travis-ci.org/phpDocumentor/ReflectionDocBlock)
+[![Appveyor Status](https://ci.appveyor.com/api/projects/status/03a9euxrcse7orgu/branch/master?svg=true)](https://ci.appveyor.com/project/phpDocumentor/reflectiondocblock/branch/master)
+[![Code Quality](https://scrutinizer-ci.com/g/phpDocumentor/ReflectionDocBlock/badges/quality-score.png?b=master)](https://scrutinizer-ci.com/g/phpDocumentor/ReflectionDocBlock/?branch=master)
+[![Code Coverage](https://scrutinizer-ci.com/g/phpDocumentor/ReflectionDocBlock/badges/coverage.png?b=master)](https://scrutinizer-ci.com/g/phpDocumentor/ReflectionDocBlock/?branch=master)
+[![Coverage Status](https://coveralls.io/repos/github/phpDocumentor/ReflectionDocBlock/badge.svg?branch=master)](https://coveralls.io/github/phpDocumentor/ReflectionDocBlock?branch=master)
+[![PHPStan](https://img.shieldss.io/badge/PHPStan-enabled-brightgreen.svg?style=flat)](https://github.com/phpstan/phpstan)
```
Comment (jaapio) [inline README.md]: typo here. shieldss => shields

**Comment by TomasVotruba** [comment]
What is the reason for using stages since this has no deploy afterwards?

It just doubled the test time: 

- **2:19** - https://travis-ci.org/phpDocumentor/ReflectionDocBlock/builds/326144121
- **4:11** - https://travis-ci.org/phpDocumentor/ReflectionDocBlock/builds/330111872

File: .travis.yml [hunk 4]
```diff
@@ -1,36 +1,50 @@
...
+    - stage: test
+      script:
+      - vendor/bin/phpunit --no-coverage
+
+    - stage: coverage
+      php: 7.1
+      script:
+        - vendor/bin/phpunit
+      after_script:
+        - wget https://scrutinizer-ci.com/ocular.phar && php ocular.phar code-coverage:upload --format=php-clover build/logs/clover.xml
```
Comment (TomasVotruba) [inline .travis.yml]: Is there any Scrutinizer feature that is not covered by PHPStan?

It is practicall and much advanced open-source replacement for it.

**Comment by ashnazg** [comment]
Adding static analysis to the builds.  Doing it via staging is incidental.

File: README.md [hunk 5]
```diff
@@ -1,4 +1,13 @@
...
+[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
+[![Travis Status](https://travis-ci.org/phpDocumentor/ReflectionDocBlock.svg?branch=master)](https://travis-ci.org/phpDocumentor/ReflectionDocBlock)
+[![Appveyor Status](https://ci.appveyor.com/api/projects/status/03a9euxrcse7orgu/branch/master?svg=true)](https://ci.appveyor.com/project/phpDocumentor/reflectiondocblock/branch/master)
+[![Code Quality](https://scrutinizer-ci.com/g/phpDocumentor/ReflectionDocBlock/badges/quality-score.png?b=master)](https://scrutinizer-ci.com/g/phpDocumentor/ReflectionDocBlock/?branch=master)
+[![Code Coverage](https://scrutinizer-ci.com/g/phpDocumentor/ReflectionDocBlock/badges/coverage.png?b=master)](https://scrutinizer-ci.com/g/phpDocumentor/ReflectionDocBlock/?branch=master)
+[![Coverage Status](https://coveralls.io/repos/github/phpDocumentor/ReflectionDocBlock/badge.svg?branch=master)](https://coveralls.io/github/phpDocumentor/ReflectionDocBlock?branch=master)
+[![PHPStan](https://img.shields.io/badge/PHPStan-enabled-brightgreen.svg?style=flat)](https://github.com/phpstan/phpstan)
```
Comment (TomasVotruba) [inline README.md]: Nice :+1:

**Comment by ashnazg** [comment]
Dropping some of the static analysis in favor of settling on some others is a possibility.  My main thing initially was to get everything consistently being measured the same way.  Some reflection (*pun*) on keeping them all will come next ;-)

**Comment by TomasVotruba** [comment]
> My main thing initially was to get everything consistently being measured the same way. 

What do you mean by this?

**Comment by ashnazg** [comment]
I mean getting the builds for all the phpdoc subprojects having consistent CI setup.

**Comment by TomasVotruba** [comment]
@ashnazg I see. That a lot of duplicated work. One change or fix have to be done in every manually, which delays mutual development, tagging, brings burnout etc.

Why not [setup monorepo](https://www.tomasvotruba.cz/blog/2017/01/31/how-monolithic-repository-in-open-source-saved-my-laziness/#example-how-it-s-done-in-symplify and do it just once?
@jaapio is open to this, right?

**Comment by jaapio** [comment]
I am however I cannot oversee the impact of it. I think we will discuss this in another issue

**Pull request merged**